src/video/wayland/SDL_waylanddyn.c

```diff
@@ -28,36 +28,54 @@
 
 #ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC
 
-typedef struct
-{
-    SDL_SharedObject *lib;
-    const char *libname;
-} waylanddynlib;
-
-static waylanddynlib waylandlibs[] = {
-    { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC },
 #ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_EGL
-    { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_EGL },
+#define SDL_WAYLAND_EGL_APPEND ,SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_EGL
+#else
+#define SDL_WAYLAND_EGL_APPEND
 #endif
 #ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_CURSOR
-    { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_CURSOR },
+#define SDL_WAYLAND_CURSOR_APPEND ,SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_CURSOR
+#else
+#define SDL_WAYLAND_CURSOR_APPEND
 #endif
 #ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_XKBCOMMON
-    { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_XKBCOMMON },
+#define SDL_WAYLAND_XKBCOMMON_APPEND ,SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_XKBCOMMON
+#else
+#define SDL_WAYLAND_XKBCOMMON_APPEND
 #endif
 #ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_LIBDECOR
-    { NULL, SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_LIBDECOR },
+#define SDL_WAYLAND_LIBDECOR_APPEND ,SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC_LIBDECOR
+#else
+#define SDL_WAYLAND_LIBDECOR_APPEND
 #endif
-    { NULL, NULL }
+
+#define SDL_WAYLAND_DYNAMIC_LIBS \
+    SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC \
+    SDL_WAYLAND_EGL_APPEND \
+    SDL_WAYLAND_CURSOR_APPEND \
+    SDL_WAYLAND_XKBCOMMON_APPEND \
+    SDL_WAYLAND_LIBDECOR_APPEND
+
+SDL_ELF_NOTE_DLOPEN(
+    "wayland",
+    "Support for Wayland video",
+    SDL_ELF_NOTE_DLOPEN_PRIORITY_REQUIRED,
```

Comment:
I think these should be `RECOMMENDED` or even `SUGGESTED`. You don't need Wayland libraries unless you are actually running on Wayland, and if packagers are OK with having a hard dependency then they can turn off `SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC`.

Reply:
I demoted them to suggested, and created a note for each required library.